interactive_demo.py

```diff
@@ -0,0 +1,200 @@
+    """
+    Add some constant values to pad a vector.
+    """
+
+    def __init__(self, num_inputs, num_outputs, output_value=0.0):
+        super().__init__()
+
+        if num_inputs > num_outputs:
+            raise ValueError('Must have more outputs than inputs')
+
+        self._difference = num_outputs - num_inputs
+        self._output_value = output_value
+
+        self._input_port = self.DeclareVectorInputPort(
+            'input', BasicVector_[float](num_inputs))
+
+        self.DeclareVectorOutputPort(
+            'output', BasicVector_[float](num_outputs), self._pad_output)
+
+    def _pad_output(self, context, output):
+        in_vec = self._input_port.Eval(context)
+        vector = [v for v in in_vec]
+        for i in range(self._difference):
+            vector.append(self._output_value)
+        output.SetFromVector(vector)
+
+
+def main():
+    builder = DiagramBuilder()
+
+    station = builder.AddSystem(ManipulationStation())
+    station.SetupClutterClearingStation()
+    station.Finalize()
+
+    # Create robot description with re-written paths
+    this_dir = os.path.abspath(os.path.dirname(__file__))
+    sdf_file_path = os.path.join(this_dir, 'iiwa14_no_collision.sdf')
+    with open(os.path.join(this_dir, 'iiwa14_no_collision.sdf.in'), 'r') as file_in:
+        with open(sdf_file_path, 'w') as file_out:
+            file_out.write(file_in.read().replace('PWD_GOES_HERE', this_dir))
+
+    joint_names = [
+        'iiwa_joint_1',
+        'iiwa_joint_2',
+        'iiwa_joint_3',
+        'iiwa_joint_4',
+        'iiwa_joint_5',
+        'iiwa_joint_6',
+        'iiwa_joint_7']
+    joints = []
+    for name in joint_names:
+        joints.append(station.get_controller_plant().GetJointByName(name))
+
+    rclpy.init()
+    node = rclpy.create_node('interactive_demo')
+
+    # Publish SDF content on robot_description topic
+    latching_qos = QoSProfile(depth=1, durability=DurabilityPolicy.TRANSIENT_LOCAL)
+    description_publisher = node.create_publisher(StringMsg, 'robot_description', qos_profile=latching_qos)
+    msg = StringMsg()
+    with open(sdf_file_path, 'r') as sdf_file:
+        msg.data = sdf_file.read()
+    description_publisher.publish(msg)
+
+    clock_system = SystemClock()
+
+    builder.AddSystem(clock_system)
+
+    # Transform broadcaster for TF frames
+    tf_broadcaster = TransformBroadcaster(node)
+
+    # Connect to system that publishes TF transforms
+    tf_system = builder.AddSystem(TFPublisher(tf_broadcaster=tf_broadcaster, joints=joints))
+    tf_buffer = Buffer(node=node)
+    tf_listener = TransformListener(tf_buffer, node)
+    server = InteractiveMarkerServer(node, 'joint_targets')
+    joint_target_system = builder.AddSystem(MovableJoints(server, tf_buffer, joints))
+
+    fk_system = builder.AddSystem(ForwardKinematics(station.get_controller_plant()))
+
+    # Blindly assume first joints in plant are the iiwa14
+    just_arm_joints = builder.AddSystem(
```

Comment:
Is this (these two lines and `PadVector`) still needed?

Reply:
Removed unused `PadVector` system 4b54925